src/edit/selection.js

```diff
@@ -65,14 +65,14 @@ class SelectionState {
   // : () → bool
   // Whether the DOM selection has changed from the last known state.
   domChanged() {
-    let sel = window.getSelection()
+    let sel = this.pm.root.getSelection()
     return sel.anchorNode != this.lastAnchorNode || sel.anchorOffset != this.lastAnchorOffset ||
       sel.focusNode != this.lastHeadNode || sel.focusOffset != this.lastHeadOffset
   }
 
   // Store the current state of the DOM selection.
   storeDOMState() {
-    let sel = window.getSelection()
+    let sel = this.pm.root.getSelection()
```

Comment:
Can this code be executed when `this.pm.root` is `null`? If so a null check may be worthwhile.